internal/dag/conditions.go

```diff
@@ -105,6 +126,9 @@ func includeMatchConditionsValid(conds []contour_api_v1.MatchCondition) error {
 		if cond.Exact != "" {
 			return fmt.Errorf("exact conditions are not allowed in includes block")
 		}
+		if cond.Regex != "" {
+			return fmt.Errorf("regex conditions are not allowed in an included HTTPProxy")
```

Comment:
I think it'd be a little more clear to use similar wording as for exact match above, since this is about not allowing regex in the include conditions themselves, not in the conditions within the included HTTPProxy's routes:
```suggestion
			return fmt.Errorf("regex conditions are not allowed in includes block")
```